pkg/virt-controller/watch/node.go

```diff
@@ -0,0 +1,245 @@
+package watch
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	v13 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+
+	v12 "kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/controller"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/log"
+)
+
+type NodeController struct {
+	clientset        kubecli.KubevirtClient
+	Queue            workqueue.RateLimitingInterface
+	nodeInformer     cache.SharedIndexInformer
+	recorder         record.EventRecorder
+	heartBeatTimeout time.Duration
+	recheckInterval  time.Duration
+}
+
+func NewNodeController(clientset kubecli.KubevirtClient, nodeInformer cache.SharedIndexInformer, recorder record.EventRecorder) *NodeController {
+	c := &NodeController{
+		clientset:        clientset,
+		Queue:            workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		nodeInformer:     nodeInformer,
+		recorder:         recorder,
+		heartBeatTimeout: 5 * time.Minute,
+		recheckInterval:  1 * time.Minute,
```

Comment:
at some point we may want to consider making this a cli option. Reason being, if there is ever an issue with these thresholds, we could solve it with a configuration change rather than requiring a new release.